### ISSUE TITLE
Bug - Application Launcher - critical axe violations in variants with icons and sections

### ISSUE THREAD

**Issue by jenny-s51** [patternfly/patternfly-react#8939]
**Describe the problem**
A clear and concise description of the problem. Which components are affected?

- https://patternfly.org/v4/components/application-launcher#with-sections-and-icons
- https://www.patternfly.org/v4/components/application-launcher#with-favorites-and-search
- https://www.patternfly.org/v4/components/application-launcher#router-link

**How do you reproduce the problem?**
Provide steps to reproduce. A [codesandbox](https://codesandbox.io/s/serverless-cherry-q9t3f ) demonstrating the problem is appreciated.

1. Open the app launcher in this example: https://patternfly.org/v4/components/application-launcher#with-sections-and-icons
2.  Run an axe scan on the current UI state
3. Note critical axe errors: Certain aria roles must contain particular children, Images must have alternate text 

**Expected behavior**
A clear and concise description of the expected behavior.

**Is this issue blocking you?**

The PF app launcher is throwing this a11y violation in ODH - certain aria roles must contain particular children

**Screenshots**
If applicable, add screenshots to help explain the issue.
<img width="1329" alt="Screenshot 2023-04-11 at 2 11 26 PM" src="https://user-images.githubusercontent.com/32821331/231252536-726a5b6f-68af-4973-b416-135fef207677.png">

<img width="1383" alt="Screenshot 2023-04-11 at 2 12 30 PM" src="https://user-images.githubusercontent.com/32821331/231252757-328d91e8-1005-4c4a-bd72-1c0f3623e951.png">

**Comment by nicolethoen** [comment]
This issue is still relevant in the composable application launcher demo (which will replace the deprecated Application Launcher in v5) http://patternfly-react-v5.surge.sh/components/menus/custom-menus#composable-application-launcher

**Comment by andrewballantyne** [comment]
@nicolethoen @jenny-s51 Given the Application Launcher is deprecated -- are we looking to just move away from it and the solution is already in PF v5's menu? Curious if this is just a matter of getting off a deprecated component or if the solution is still waiting on PF to solve it somehow?

**Comment by jenny-s51** [comment]
@andrewballantyne yes, we are looking to move away from the Application Launcher and after testing on the product side, the original issues no longer apply. 

However, in the PF workspace...

Axe is still showing critical a11y errors in the [new v5 "Custom Menu" implementation of the Application Launcher](https://www.patternfly.org/components/menus/custom-menus/#application-launcher-menu) - seems to be related to favorites logic/components and the `icon` component missing `alt` text. When favorites are removed and alt text is added to the PF icon, Axe tests pass.

<img width="800" alt="Screenshot 2023-12-11 at 11 12 12 AM" src="https://github.com/patternfly/patternfly-react/assets/32821331/a5c20176-78f8-42d2-b480-1ac713d024ec">

Going through the demos, I'm seeing several critical Axe violations throughout the custom menus and these could all be worth fixing together, perhaps as a standalone issue. WDYT @tlabaj @nicolethoen ?

**Comment by nicolethoen** [comment]
@jenny-s51 sounds good to me! let's capture what we can in a clear issue and we can do a sweeping docs update if we have a bad habit we need to address in our Menu docs

**Comment by jenny-s51** [comment]
Closing in favor of #9968
